parsers/AX_NEW.py

```diff
@@ -0,0 +1,208 @@
+from datetime import datetime, timedelta
+from logging import Logger, getLogger
+from re import findall
+from typing import List, Optional
+
+from bs4 import BeautifulSoup
+from pytz import timezone
+from requests import Response, Session
+
+from .lib.exceptions import ParserException
+
+IFRAME_URL = "https://grafik.kraftnat.ax/grafer/tot_inm_24h_15.php"
+TIME_ZONE = "Europe/Mariehamn"
+SOURCE = "kraftnat.ax"
+
+
+def fetch_data(session: Session, logger: Logger):
+    """Fetch data from the iFrame."""
+
+    response: Response = session.get(IFRAME_URL)
+    response.raise_for_status()
+
+    soup = BeautifulSoup(response.text, "html.parser").find_all("script")
+    result_time_series = findall(r"data: \[(.*?)\]", str(soup))
+    if len(result_time_series) != 3:
+        raise ParserException(
+            "AX.py",
+            "Did not find the expected amount of results. Check if the website has changed.",
+        )
+    time_series: List = result_time_series[0].split(",")
+    raw_data: List[str] = findall(r"data:\[(.*?)\]", str(soup))
+    if len(raw_data) != 6:
+        raise ParserException(
+            "AX.py",
+            "The raw data did not match the expected format. Check if the website has changed.",
+        )
+    for raw in raw_data:
+        if len(raw.split(",")) != len(time_series):
+            raise ParserException(
+                "AX.py",
+                "The raw data did not match the length of the the time series. Check if the website has changed.",
+            )
+    data_list = []
+    for time, sweden, alink, fossil, gustavs, wind, consumption in zip(
+        time_series,
+        raw_data[0].split(","),
+        raw_data[1].split(","),
+        raw_data[2].split(","),
+        raw_data[3].split(","),
+        raw_data[4].split(","),
+        raw_data[5].split(","),
+    ):
+        data_list.append(
+            {
+                "time": str(time.replace('"', "")),
+                "sweden": float(sweden),
+                "alink": float(alink),
+                "fossil": float(fossil),
+                "gustavs": float(gustavs),
+                "wind": float(wind),
+                "consumption": float(consumption),
+            }
+        )
+    return data_list
+
+
+def formated_data(
```

Comment:
This should probably live in the shared lib folder at some point.
It's probably ok for now to have it here, but I feel that all parsers would benefit from sharing such a function.

Reply:
This function in particular probably won't work well shared but we could make a shared function that formats data to our different return types (these could also include some basic type checks if we want).